### PR TITLE
feat: add documentation for user group management

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Issue: https://github.com/camunda/tasklist/issues/3866

Add the documentation for User restrictions on Self-Managed document. 

## When should this change go live?

This change goes live on release of 8.4.0-alpha2, on the week of December 2th

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [ ] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by christinaausley — APPROVED** [review]

File: docs/self-managed/tasklist-deployment/tasklist-authentication.md [hunk 1]
```diff
@@ -144,6 +149,36 @@ Take the `access_token` value from the response object and store it as your toke
...
+
+#### Candidate users
+
+- Tasks will be filtered based on candidate users specified for each task.
+- If a task is configured with candidate users, only those users will see the task in their task list.
+
+#### Assignees
+
+- Tasks assigned to a specific user will only be visible to that assigned user and to the users that belong to the candidate groups/users associated with the task.
+- If the logged-in user is the assignee for a task, the task will be displayed in their task list independent of whether the user is on candidate groups/users.
```
Comment (christinaausley) [inline docs/self-managed/tasklist-deployment/tasklist-authentication.md:172]: > If the logged-in user is the assignee for a task, the task will be displayed in their task list independent of whether the user is on candidate groups/users.

Do you mean: if the active user is assigned to a task, the task is displayed in their task list regardless of their candidate group or user status.?

**Review by christinaausley — APPROVED** [review]
One comment for clarity 👍 Nice work!

**Comment by christinaausley** [comment]
@nathansandi Can I remove the `hold` label and merge here, or are you planning to make additional adjustments?

**Comment by nathansandi** [comment]
> @nathansandi Can I remove the `hold` label and merge here, or are you planning to make additional adjustments?

I believe you can merge it

**Pull request merged**